doc/pmreorder/pmreorder.1.md

```diff
@@ -0,0 +1,242 @@
+collection: pmreorder
+header: PMDK
+date: pmreorder version 1.5
+...
+
+[comment]: <> (Copyright 2018, Intel Corporation)
+
+[comment]: <> (Redistribution and use in source and binary forms, with or without)
+[comment]: <> (modification, are permitted provided that the following conditions)
+[comment]: <> (are met:)
+[comment]: <> (    * Redistributions of source code must retain the above copyright)
+[comment]: <> (      notice, this list of conditions and the following disclaimer.)
+[comment]: <> (    * Redistributions in binary form must reproduce the above copyright)
+[comment]: <> (      notice, this list of conditions and the following disclaimer in)
+[comment]: <> (      the documentation and/or other materials provided with the)
+[comment]: <> (      distribution.)
+[comment]: <> (    * Neither the name of the copyright holder nor the names of its)
+[comment]: <> (      contributors may be used to endorse or promote products derived)
+[comment]: <> (      from this software without specific prior written permission.)
+
+[comment]: <> (THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS)
+[comment]: <> ("AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT)
+[comment]: <> (LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR)
+[comment]: <> (A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT)
+[comment]: <> (OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,)
+[comment]: <> (SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT)
+[comment]: <> (LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,)
+[comment]: <> (DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY)
+[comment]: <> (THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT)
+[comment]: <> ((INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE)
+[comment]: <> (OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.)
+
+[comment]: <> (pmreorder.1 -- man page for daxio)
+
+[NAME](#name)<br />
+[SYNOPSIS](#synopsis)<br />
+[DESCRIPTION](#description)<br />
+[OPTIONS](#options)<br />
+[ENGINES](#engines)<br />
+[INSTRUMENTALIZATION](#instrumentalization)<br />
+[PMEMCHECK STORE LOG](#pmemcheck-store-log)<br />
+[EXAMPLE](#example)<br />
+[SEE ALSO](#see-also)<br />
+
+
+# NAME #
+
+**pmreorder** -- performs persistent consisteny check
+		 using store reordering mechanism
+
+
+# SYNOPSIS #
+
+```
+$ python pmreorder <options>
+```
+
+
+# DESCRIPTION #
+
+The pmreorder tool is a collection of python scripts designed
+to parse and replay operations logged by pmemcheck -
+a persistent memory checking tool.
+
+Pmreorder performs store reordering between prersistent
+memory barriers - a sequence of flush-fence operations.
+It uses the consistency checking routine provided in the
+command line options to check whether the files are in a
+consistent state.
+
+Seeing as the combination of valgrind and the multiple store
```

Comment:
Considering that logging of operations under valgrind, and its then reply and reorder, is very time consuming, it is recommended to use as few stores as possible in test workloads.